diff --git a/Android.bp b/Android.bp
index f36d4cd..4c0da1d 100644
--- a/Android.bp
+++ b/Android.bp
@@ -42,7 +42,10 @@ rust_library {
     crate_name: "libsqlite3_sys",
     cargo_env_compat: true,
     cargo_pkg_version: "0.22.2",
-    srcs: ["src/lib.rs"],
+    srcs: [
+        "src/lib.rs",
+        ":libsqlite3_bindgen",
+    ],
     edition: "2018",
     features: [
         "default",
@@ -50,6 +53,7 @@ rust_library {
         "pkg-config",
         "vcpkg",
     ],
+    shared_libs: ["libsqlite"],
     apex_available: [
         "//apex_available:platform",
         "//apex_available:anyapex",
@@ -62,7 +66,10 @@ rust_test {
     crate_name: "libsqlite3_sys",
     cargo_env_compat: true,
     cargo_pkg_version: "0.22.2",
-    srcs: ["src/lib.rs"],
+    srcs: [
+        "src/lib.rs",
+        ":libsqlite3_bindgen",
+    ],
     test_suites: ["general-tests"],
     auto_gen_config: true,
     test_options: {
@@ -75,6 +82,7 @@ rust_test {
         "pkg-config",
         "vcpkg",
     ],
+    shared_libs: ["libsqlite"],
 }
 
 rust_bindgen {
