--- Android.bp
+++ Android.bp
@@ -45,6 +45,7 @@
     cargo_env_compat: true,
     cargo_pkg_version: "0.22.2",
     crate_root: "src/lib.rs",
+    srcs: [":libsqlite3_bindgen"],
     edition: "2018",
     features: [
         "default",
@@ -52,6 +53,7 @@
         "pkg-config",
         "vcpkg",
     ],
+    shared_libs: ["libsqlite"],
     apex_available: [
         "//apex_available:platform",
         "//apex_available:anyapex",
@@ -65,6 +67,7 @@
     cargo_env_compat: true,
     cargo_pkg_version: "0.22.2",
     crate_root: "src/lib.rs",
+    srcs: [":libsqlite3_bindgen"],
     test_suites: ["general-tests"],
     auto_gen_config: true,
     test_options: {
@@ -77,6 +80,7 @@
         "pkg-config",
         "vcpkg",
     ],
+    shared_libs: ["libsqlite"],
 }
 
 rust_bindgen {
